src/CircleEditor/CircleEditorDocument.ts

```diff
@@ -191,4 +195,333 @@ export class CircleEditorDocument extends Disposable implements vscode.CustomDoc
       }
     };
   }
+
+  /**
+   * Guess data's type (int or float)
+   */
+  private guessExactType(n: any) {
+    if (Number(n) % 1 === 0) {
```

Comment:
Is `Number` is correct, not `number`? 
And it would better to add assert when `n` is not a number :)

Reply:
OK I'll add assert when `n` is not a number :)